#!/usr/bin/env python
PACKAGE = "rnw_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cage_center_x", double_t, 0, "cage_center_x", 0, -10, 10)
gen.add("cage_center_y", double_t, 0, "cage_center_y", 0, -10, 10)
gen.add("cage_center_z", double_t, 0, "cage_center_z", 0, -10, 10)

gen.add("base_center_x", double_t, 0, "base_center_x", 0, -10, 10)
gen.add("base_center_y", double_t, 0, "base_center_y", 0, -10, 10)
gen.add("base_center_z", double_t, 0, "base_center_z", 0, -10, 10)

gen.add("G1_x", double_t, 0, "G1_x", 0, -10, 10)
gen.add("G1_y", double_t, 0, "G1_y", 0, -10, 10)
gen.add("G1_z", double_t, 0, "G1_z", 0, -10, 10)

gen.add("G2_x", double_t, 0, "G2_x", 0, -10, 10)
gen.add("G2_y", double_t, 0, "G2_y", 0, -10, 10)
gen.add("G2_z", double_t, 0, "G2_z", 0, -10, 10)

gen.add("Tip_x", double_t, 0, "Tip_x", 0, -10, 10)
gen.add("Tip_y", double_t, 0, "Tip_y", 0, -10, 10)
gen.add("Tip_z", double_t, 0, "Tip_z", 0, -10, 10)

gen.add("uav_pt_x", double_t, 0, "Calculate position in UAV's marker frame", 0, -10, 10)
gen.add("uav_pt_y", double_t, 0, "Calculate position in UAV's marker frame", 0, -10, 10)
gen.add("uav_pt_z", double_t, 0, "Calculate position in UAV's marker frame", 0, -10, 10)

gen.add("cone_pt_x", double_t, 0, "Calculate position in cone's marker frame", 0, -10, 10)
gen.add("cone_pt_y", double_t, 0, "Calculate position in cone's marker frame", 0, -10, 10)
gen.add("cone_pt_z", double_t, 0, "Calculate position in cone's marker frame", 0, -10, 10)

exit(gen.generate(PACKAGE, "rnw_ros", "MocapCalib"))